packages/langium/test/lsp/completion-provider.test.ts

```diff
@@ -202,3 +202,75 @@ describe('Path import completion', () => {
         });
     });
 });
+
+describe('Completion in data type rules', () => {
+
+    test('Can perform completion for fully qualified names', async () => {
+        const grammar = `
+        grammar FQNCompletionTest
+
+        entry Model:
+            (persons+=Person | greetings+=Greeting)*;
+
+        Person:
+            'person' name=FQN;
+
+        Greeting:
+            'Hello' person=[Person:FQN] '!';
+
+        FQN returns string: ID ('.' ID)*;
```

Comment:
I am wondering,

1. what kind of string value will be FQN for `abc  . /*comment*/ xyz`
2. whether it would be good to add test cases which handle edge cases like
  * having whitespace and comments in-between
  * also test the case when the cursor is directly(!) in front of or behind of the dot '.'

Reply:
1. `abc  . /*comment*/ xyz` will be turned into `abc.xyz`, as hidden tokens aren't respected while building data type rule values.
2. I've added a test for comments and whitespace within FQN. They should result in no completion items being returned. As for the other one, we already have a test for right in front of the period. Having the cursor after the dot is the same edge case (i.e. the cursor is between two tokens of the FQN).